cmd/az-analyze/cmd/azva_test.go

```diff
@@ -0,0 +1,345 @@
+				expect := []AzvaResource{azvaResource_pod1, azvaResource_pod2}
+
+				for i := 0; i < len(result); i++ {
+					require.Equal(t, result[i].ResourceType, expect[i].ResourceType)
+					require.Equal(t, result[i].Namespace, expect[i].Namespace)
+					require.Equal(t, result[i].Name, expect[i].Name)
+					require.Equal(t, int(result[i].Age / time.Second), int(expect[i].Age / time.Second))
+					require.Equal(t, result[i].RequestRole, expect[i].RequestRole)
+					require.Equal(t, result[i].Role, expect[i].Role)
+				}
+			},
+		},
+	}
+
+	for _, test := range tests {
+		tt := test
+		t.Run(tt.description, func(t *testing.T) {
+			tt.verifyFunc()
+		})
+	}
+}
+
+func TestGetAzVolumeAttachementsByNode(t *testing.T) {
+	fakeClientsetAzDisk := NewTestAzDiskClientset()
+
+	tests := []struct {
+		description string
+		verifyFunc  func()
+	}{
+		{
+			description: "specified node name attached one pvc",
+			verifyFunc: func() {
+				result := GetAzVolumeAttachementsByNode(fakeClientsetAzDisk, "test-node-0")
+				expect := []AzvaResource{azvaResource_node1}
+
+				for i := 0; i < len(result); i++ {
+					require.Equal(t, result[i].ResourceType, expect[i].ResourceType)
+					require.Equal(t, result[i].Namespace, expect[i].Namespace)
+					require.Equal(t, result[i].Name, expect[i].Name)
+					require.Equal(t, int(result[i].Age / time.Second), int(expect[i].Age / time.Second))
+					require.Equal(t, result[i].RequestRole, expect[i].RequestRole)
+					require.Equal(t, result[i].Role, expect[i].Role)
+				}
+			},
+		},
+		{
+			description: "specified node name attached more than one pvc",
+			verifyFunc: func() {
+				result := GetAzVolumeAttachementsByNode(fakeClientsetAzDisk, "test-node-1")
+				expect := []AzvaResource{azvaResource_node2, azvaResource_node3}
+
+				for i := 0; i < len(result); i++ {
+					require.Equal(t, result[i].ResourceType, expect[i].ResourceType)
+					require.Equal(t, result[i].Namespace, expect[i].Namespace)
+					require.Equal(t, result[i].Name, expect[i].Name)
+					require.Equal(t, int(result[i].Age / time.Second), int(expect[i].Age / time.Second))
+					require.Equal(t, result[i].RequestRole, expect[i].RequestRole)
+					require.Equal(t, result[i].Role, expect[i].Role)
+				}
+			},
+		},
+	}
+
+	for _, test := range tests {
+		tt := test
+		t.Run(tt.description, func(t *testing.T) {
+			tt.verifyFunc()
+		})
+	}
+}
+
+func TestGetAzVolumeAttachementsByZone(t *testing.T) {
+	fakeClientsetK8s := NewTestK8sClientset()
+	fakeClientsetAzDisk := NewTestAzDiskClientset()
+
+	tests := []struct {
+		description string
+		verifyFunc  func()
+	}{
+		{
+			description: "specified zone name with one node",
+			verifyFunc: func() {
+				result := GetAzVolumeAttachementsByZone(fakeClientsetK8s, fakeClientsetAzDisk, "eastus-0")
+				expect := []AzvaResource{azvaResource_zone1}
+
+				for i := 0; i < len(result); i++ {
+					require.Equal(t, result[i].ResourceType, expect[i].ResourceType)
```

Comment:
We could take this common part in a function and reuse it across the test functions.